taggui/models/image_list_model.py

```diff
@@ -241,21 +242,28 @@ def is_image_in_scope(self, scope: Scope | str, image_index: int,
             return self.image_list_selection_model.isSelected(proxy_index)
 
     def get_text_match_count(self, text: str, scope: Scope | str,
-                             whole_tags_only: bool) -> int:
+                             whole_tags_only: bool, use_regex: bool) -> int:
         """Get the number of instances of a text in all captions."""
         match_count = 0
         for image_index, image in enumerate(self.images):
             if not self.is_image_in_scope(scope, image_index, image):
                 continue
             if whole_tags_only:
-                match_count += image.tags.count(text)
+                if use_regex:
+                    for tag in image.tags:
+                        match_count += len(re.findall(text, tag))
```

Comment:
This is wrong: this counts the number of matches inside each tag, but instead, it should count the number of tags that are complete matches for the regex. I will fix it.

Reply:
When "Whole tags only" is checked, entire tags get replaced, and not parts of tags. The regex implementation should follow this original behavior.